javagen/src/main/java/com/azure/autorest/model/clientmodel/ListType.java

```diff
@@ -30,10 +30,19 @@ public void addImportsTo(Set<String> imports, boolean includeImplementationImpor
         super.addImportsTo(imports, includeImplementationImports);
     }
 
+    @Override
     public String validate(String expression) {
-        String elementValidation = getElementType().validate("e");
+        return validate(expression, 0);
+    }
+
+    @Override
+    public String validate(String expression, int depth) {
+        String var = depth == 0 ? "e" : "e" + depth;
+        String elementValidation = getElementType() instanceof GenericType
+                ? ((GenericType) getElementType()).validate(var, ++depth)
+                : getElementType().validate(var);
```

Comment:
To generate this `compositeIndexes().forEach(e -> e.forEach(e1 -> e1.validate()));`

Probably not a good fix. Any suggestion?